ui/chatview.go

```diff
@@ -676,7 +676,8 @@ func parseCustomEmojis(text string) string {
 
 	customEmojiMatches := customEmojiRegex.FindAllStringSubmatch(messageText, -1)
 	for _, match := range customEmojiMatches {
-		url := "https://cdn.discordapp.com/emojis/" + match[2]
+		custom_emoji_code := strings.Split(match[0], ":")[1]
```

Comment:
This code contains a bug. It should also be signaled, that an emoji is animated. Add a unit test that contains a custom emoji. `Hello <a:owo:123>`

The problem is, that you split by `:` which splits it into `isAnimated`, `name` and `ID`.

What you could do instead, is to just make a new matchgroup.

```go
customEmojiRegex           = regexp.MustCompile("(?sm)(.?)<a?:.+?:(\\d+)>(.?)")
```

```go
customEmojiRegex           = regexp.MustCompile("(?sm)(.?)<(a?:.+?):(\\d+)>(.?)")
```

And instead of using `strings.Split`, you can just use the matcher group `2`. The groups that were `2` and `3` before will now be `3` and `4`. Be sure to correct that as well.

Reply:
Fixed it, you can now tell by the `a:` if an emoji is animated or not :p